enhancements/installer/connected-onprem-assisted-installer.md

```diff
@@ -0,0 +1,303 @@
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+Since the on-premise assisted installer is intended for short-term use just to get
+a cluster installed, there is nothing stateful to upgrade or downgrade.
+
+The user will retrieve a newer version, or potentially an older version if
+older versions are offered, by downloading it from cloud.redhat.com.
+
+### Version Skew Strategy
+
+How will the component handle version skew with other components?
+What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- During an upgrade, we will always have skew among components, how will this impact your work?
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI, CRI
+  or CNI may require updating that component before the kubelet.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+Running the assisted installer on-premise is not as simple of an experience as
+utilizing the service from cloud.redhat.com. It requires more work from the
+user prior to beginning a cluster installation.
+
+## Alternatives
+
+### Containers Instead of ISO
+
+Instead of distributing the assisted installer as a live ISO, it could be distributed
+as a podman-compatible Pod definition that runs containers directly on the
+user's machine. The user would copy a `podman` command from cloud.redhat.com and run it
+locally.
+
+Advantages include:
+* no need to support a number of potential virtualization platforms
+* exposing the assisted installer's API service from a container may be easier in many cases than exposing it from a VM
+* in the future the live ISO can run on one of the hosts and serve the bootstrap role. This requires an OS on that host, which the live ISO provides.
+
+Downsides:
```

Comment:
Also ISO will need to be downloaded, right?

Reply:
It needs to be downloaded either way, and I'm not sure that either results in a substantially different user experience.